botwo/botwo.py

```diff
@@ -0,0 +1,188 @@
+        operations = [func for func in dir(self.default_client) if
+                      (callable(getattr(self.default_client, func)) and not func.startswith('_'))]
+        for operation_name in operations:
+            if operation_name in SPECIAL_METHODS:
+                op_dict[operation_name] = self._create_special_method(operation_name)
+            elif operation_name in COPY_METHODS:
+                op_dict[operation_name] = self._create_copy_method(operation_name)
+            else:
+                op_dict[operation_name] = self._create_api_method(operation_name)
+        return op_dict
+
+    def _create_api_method(self, operation_name):
+        def _api_call(*args, **kwargs):
+            if args:
+                raise TypeError("%s() only accepts keyword arguments." % operation_name)
+            client_to_call = self.default_client
+            if kwargs.get("Bucket"):  # bucket operation
+                full_path = kwargs["Bucket"] + "/" + kwargs["Key"]
+                res = self._route_params(full_path, kwargs, is_copy=False)
+                client_to_call = res[0]
+                api_params = res[1]
+
+            return getattr(client_to_call, operation_name)(**api_params)
+
+        _api_call.__name__ = str(operation_name)
+        return _api_call
+
+    def _create_copy_method(self, operation_name):
+        def _api_call(*args, **kwargs):
+            if args:
+                raise TypeError("%s() only accepts keyword arguments." % operation_name)
+            client_to_call_source = self.default_client
+            client_to_call_dest = self.default_client
+            if kwargs.get("CopySource"):  # copy operation
+                res = self._route_params(kwargs["CopySource"], kwargs, is_copy=True)
+                client_to_call_source = res[0]
+                api_params = res[1]
+
+            if api_params.get("Bucket"):
+                full_dest_path = api_params["Bucket"] + "/" + api_params["Key"]
+                res = self._route_params(full_dest_path, api_params, is_copy=False)
+                client_to_call_dest = res[0]
+                api_params = res[1]
+
+            if client_to_call_source != client_to_call_dest:
+                raise ValueError("client source and client destination are different")
+
+            return getattr(client_to_call_source, operation_name)(**api_params)
+
+        _api_call.__name__ = str(operation_name)
+        return _api_call
+
+    def _create_special_method(self, operation_name):
+        def _paginator_api_call(*args, **kwargs):
+            if args:
+                raise TypeError("%s() only accepts keyword arguments." % operation_name)
+            return Paginatwo(self.mapping, kwargs, self.client1, self.client2, self.client1_name, self.client2_name)
+
+        def _delete_objects_api_call(*args, **kwargs):
+            if args:
+                raise TypeError("%s() only accepts keyword arguments." % operation_name)
+            # TODO
```

Comment:
```suggestion
            # TODO(eden)
```
or
```suggestion
            # TODO(ISSUE NUMBER)
```

Can we raise an exception here?  Returning `None` makes the user think that the call succeeded.

Reply:
I implemented it.